vector/src/main/scala/geotrellis/vector/voronoi/Delaunay.scala

```diff
@@ -0,0 +1,324 @@
+    isCCW(p, trans(e.src), trans(e.vert))
+
+  def isLeftOf[T](e: HalfEdge[Int,T], p: Int)(implicit trans: Int => Point) = 
+    isCCW(trans(p), trans(e.src), trans(e.vert))
+
+  def inCircle(abc: (Point, Point, Point), d: Point): Boolean = {
+    abc match {
+      case (a,b,c) =>
+        det3(a.x-d.x, a.y-d.y, pow(a.x-d.x,2) + pow(a.y-d.y,2),
+             b.x-d.x, b.y-d.y, pow(b.x-d.x,2) + pow(b.y-d.y,2),
+             c.x-d.x, c.y-d.y, pow(c.x-d.x,2) + pow(c.y-d.y,2)) > 0
+    }
+  }
+
+  def inCircle(abc: (Int, Int, Int), di: Int)(implicit trans: Int => Point): Boolean = {
+    abc match {
+      case (ai,bi,ci) => {
+        val a = trans(ai)
+        val b = trans(bi)
+        val c = trans(ci)
+        val d = trans(di)
+        det3(a.x-d.x, a.y-d.y, pow(a.x-d.x,2) + pow(a.y-d.y,2),
+             b.x-d.x, b.y-d.y, pow(b.x-d.x,2) + pow(b.y-d.y,2),
+             c.x-d.x, c.y-d.y, pow(c.x-d.x,2) + pow(c.y-d.y,2)) > 0
+      }
+    }
+  }
+
+  def circleCenter(a: Point, b: Point, c: Point): Point = {
+    val d = 2.0 * det3(a.x,a.y,1.0, b.x,b.y,1.0, c.x,c.y,1.0)
+    val h = det3(a.x*a.x+a.y*a.y,a.y,1.0, b.x*b.x+b.y*b.y,b.y,1.0, c.x*c.x+c.y*c.y,c.y,1.0) / d
+    val k = det3(a.x,a.x*a.x+a.y*a.y,1.0, b.x,b.x*b.x+b.y*b.y,1.0, c.x,c.x*c.x+c.y*c.y,1.0) / d
+    Point(h,k)
+  }
+
+  def circleCenter(ai: Int, bi: Int, ci: Int)(implicit trans: Int => Point): Point = {
+    val a = trans(ai)
+    val b = trans(bi)
+    val c = trans(ci)
+    val d = 2.0 * det3(a.x,a.y,1.0, b.x,b.y,1.0, c.x,c.y,1.0)
+    val h = det3(a.x*a.x+a.y*a.y,a.y,1.0, b.x*b.x+b.y*b.y,b.y,1.0, c.x*c.x+c.y*c.y,c.y,1.0) / d
+    val k = det3(a.x,a.x*a.x+a.y*a.y,1.0, b.x,b.x*b.x+b.y*b.y,1.0, c.x,c.x*c.x+c.y*c.y,1.0) / d
+    Point(h,k)
+  }
+
+  def isDelaunayEdge[T](e: HalfEdge[Int,T])(implicit trans: Int => Point): Boolean = {
+    // Predicated on the fact that if an edge is Delaunay, then for a
+    // point, A, to the left of edge (X,Y), and a point, B, to the
+    // right of (X,Y), A may not be in the circle defined by points X,
+    // Y, and B.
+    val a = trans(e.next.vert)
+    val b = trans(e.flip.next.vert)
+    val x = trans(e.flip.vert)
+    val y = trans(e.vert)
+    !inCircle((a,x,y),b)
+  }
+}
+
+case class Delaunay(verts: Array[Point]) {
+  val triangles = Map.empty[(Int,Int,Int),HalfEdge[Int,Point]]
+  val faceIncidentToVertex = Map.empty[Int,HalfEdge[Int,Point]]
+
+  private def insertTriangle(vs: (Int,Int,Int), e: HalfEdge[Int,Point]): Map[(Int,Int,Int),HalfEdge[Int,Point]] = {
+    val idx = vs match {
+      case (a,b,c) if (a<b && a<c) => (a,b,c)
+      case (a,b,c) if (b<a && b<c) => (b,c,a)
+      case (a,b,c) => (c,a,b)
+    }
+    //println(s"Inserting $e as $vs [using $idx as key]")
+    faceIncidentToVertex += (e.vert -> e, e.next.vert -> e.next, e.next.next.vert -> e.next.next)
+    //println(s"Adding to faceIncidentToVertex: ${e.vert -> e}, ${e.next.vert -> e.next}, ${e.next.next.vert -> e.next.next}")
```

Comment:
Debug printlns should be removed. If there's a specific reason for it to be around (e.g. this is *always* the information that is needed when this breaks, so kept for perpetuity) then it should be commented on and kept, but by default all debug lines and comments should be removed.